DLL_Project/Detours/JobGiver_GetFood.cs

```diff
@@ -71,13 +71,21 @@ protected override Job TryGiveJob( Pawn pawn )
                 }
                 if( foodSource is Building_AutomatedFactory )
                 {
+                    //Fixed the exepction from passing a Factory into the WillIngestStackCountOf function.
+                    //However, truely fixing this will require detouring FoodUtilities to get proper functionality.
                     var FS = foodSource as Building_AutomatedFactory;
-                    if( foodDef == null )
+                    foodDef = FS.BestProduct(FoodSynthesis.IsMeal, FoodSynthesis.SortMeal);
+                    if( !FS.HasEnoughResourcesInHoppersFor(foodDef) )
                     {
                         //CCL_Log.Message( string.Format( "Hopper for {0} needs filling", foodSource.ThingID ) );
                         hopperNeedsFilling = true;
                         hopper = FS.AdjacentReachableHopper( pawn );
                     }
+                    //TODO I think this is broken for factories. The code does not confirm there's material to fill the hopper available.
+                    //Not an issue with nutrient paste, since if you can't fill a nutrient paste hopper you're probably starving anyway.
+                    //But if you have a factory that makes Lavish Meals from meat. Colonists will wander around starving
+                    //trying to get meat even if there's fine meals in storage. I think anyway.
+                    //Also, current code would ignore a Fine meal factory if the Lavish meal factory has unreachable hoppers.

```

Comment:
TryFindBestFoodSourceFor() works by calling several methods depending on the situation.  The situation we care about is when it calls BestFoodSourceOnMap() which has been detoured to handle factories.  Best have your IDE find all references to Building_AutomatedFactory.  That will show you all the detours, etc you need which are already in place but will need updating for this and every alpha since they were first implemented (back in ~A12 when factories were added).